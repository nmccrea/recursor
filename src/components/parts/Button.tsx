import styled from "styled-components"
import { colorPrimary, colorDanger } from "../../styles/theme/colors"
import { borderRadius } from "../../styles/utilities/borderRadius"
import {
  shadowSmall,
  shadowMedium,
  shadowLarge,
} from "../../styles/utilities/shadow"

/**
 * A generic button element with visual feedback on hover and click.
 */
export const Button = styled.button`
  ${borderRadius}
  ${shadowSmall}

  padding: 6px 16px;

  background-color: lightgray;

  color: white;
  letter-spacing: 0.03em;
  line-height: 1.75;
  text-transform: uppercase;

  &:hover {
    ${shadowMedium}
    filter: brightness(95%);
  }

  &:active {
    ${shadowLarge}
    filter: brightness(100%);
  }
`

/**
 * A button for primary, most common, or expected actions.
 */
export const ButtonPrimary = styled(Button)`
  background-color: ${colorPrimary};
`

/**
 * A button for destructive actions.
 */
export const ButtonDanger = styled(Button)`
  background-color: ${colorDanger};
`
