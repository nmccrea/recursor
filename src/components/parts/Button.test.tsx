import React from "react"
import { render } from "@testing-library/react"
import { Button, ButtonPrimary, ButtonDanger } from "./Button"

describe("`<Button />`", () => {
  it("renders correctly", () => {
    const { container } = render(<Button />)

    expect(container).toMatchSnapshot()
  })
})

describe("`<ButtonPrimary />`", () => {
  it("renders correctly", () => {
    const { container } = render(<ButtonPrimary />)

    expect(container).toMatchSnapshot()
  })
})

describe("`<ButtonDanger />`", () => {
  it("renders correctly", () => {
    const { container } = render(<ButtonDanger />)

    expect(container).toMatchSnapshot()
  })
})
