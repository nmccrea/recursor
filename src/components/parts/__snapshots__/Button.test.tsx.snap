// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`<Button />\` renders correctly 1`] = `
<div>
  <button
    class="sc-AxjAm iGplDk"
  />
</div>
`;

exports[`\`<ButtonDanger />\` renders correctly 1`] = `
<div>
  <button
    class="sc-AxjAm sc-AxiKw cgflug"
  />
</div>
`;

exports[`\`<ButtonPrimary />\` renders correctly 1`] = `
<div>
  <button
    class="sc-AxjAm sc-AxirZ CUWHb"
  />
</div>
`;
